tools/refseq_masher/contains.xml

```diff
@@ -0,0 +1,256 @@
+<tool id="refseq_masher_contains" name="RefSeq Masher Contains" version="0.1.0">
+  <description>
+    Find NCBI RefSeq Genomes contained in your sequences
+  </description>
+  <requirements>
+    <requirement type="package" version="2.0">mash</requirement>
+    <requirement type="package" version="0.1.0">refseq_masher</requirement>
```

Comment:
For the refseq_masher itself, could you maybe order by identity and median_multiplicity?  I noticed for some of my results, that they all have identity of `1.0`, but median multiplicity is unsorted.   I think this is important especially in the mode of only displaying the top **n** results.

Reply:
Also, is it possible to have refseq_masher accept paired-end data as well as single-end?

Reply:
Actually, I think you may already be sorting by both identity and multiplicity.  I think I just read my test report wrong.

Reply:
Sounds like an issue with the fastq grouping function since it should work with a mix of fasta files, fastq files, directories with fasta/fastq files. Right now it groups fastqs by stripping away `_\d`, so it would only work for files that have EBI-like naming (e.g. `SRR1234_1.fastq.gz`, `SRR1234_2.fastq.gz`). I'll change the rules for grouping if naming doesn't match that convention. I could try grouping by ID (maybe a regex like `@(\S+)\s.*`) in the first line in each fastq file. It wouldn't allow for pooling of cross run samples into one Mash dist/screen command. What do you think?

Reply:
I did not realize it grouped paired-end files together actually.  I guess some of the files I have are named like `ASM-67_S1_L001_R1_001.fastq` and it's showing them as separate results.  Files could also be named something like `ASM_67_S1_L001_R1_001.fastq` too.  Maybe parse in all file names and see if they differ by `_R1/_R2` or `_1/_2`?  Also perhaps an option `--no-group-fastqs` or something to disable grouping?

For the Galaxy tool, I would like to have support for paired-end dataset collections (http://planemo.readthedocs.io/en/latest/writing_advanced.html#processing-pairs) since that's what IRIDA uses to pass data into Galaxy.  Is this possible along with your current input method?

Reply:
Grouping by first line in each fastq file is probably not necessary.  I think just by file name is okay.

Reply:
Or, maybe instead of `--no-group-fastqs`, have an options to either (1) treat all fastq files as if from sample sample, (2) attempt to pair off fastq files in list and treat rest as from different samples, or (3) treat every individual fastq file as from a different sample.

This would let people pick how to interpret the files they pass to `mash`.

Reply:
I'm probably getting too fancy with my file handling. I just wanted to support flexible input file specification by the user through the command-line. 

So allow users to specify the following?

- one or multiple single datasets (fasta, fastq)
- one or multiple paired datasets (fastq)
- `data_collection` of any of the above (list of single/paired datasets)


For user-defined grouping, I could add an option like `--input <NAME> <PATH>` where distinct filepaths are grouped on `NAME`. So in the tool wrapper it would look something like:

```
#for $name, $input in $input_collection.items()
--input $name $input
#end for
```

Reply:
Fixed in 1bf1be8 - input file handling should be much more robust and flexible.

Reply:
Awesome, thanks.